Microsoft.Azure.Cosmos.Encryption/src/EncryptionCosmosDiagnostics.cs

```diff
@@ -86,5 +86,74 @@ public override string ToString()
 
             return stringWriter.ToString();
         }
+
+#if SDKPROJECTREF
+
+        public EncryptionCosmosDiagnostics(ITrace trace)
+        {
+            if (trace == null)
+            {
+                throw new ArgumentNullException(nameof(trace));
+            }
+
+            // Need to set to the root trace, since we don't know which layer of the stack the response message was returned from.
+            ITrace rootTrace = trace;
+            while (rootTrace.Parent != null)
+            {
+                rootTrace = rootTrace.Parent;
+            }
+
+            this.Value = rootTrace;
+        }
+        
+        public ITrace Value { get; }
+
+        public override DateTime GetStartTimeUtc()
```

Comment:
EncryptionCosmosDiagnostics is just a wrapper around the original CosmosDiagnostics object. So instead of duplicating the logic just call original method like GetContactedRegion.

```suggestion
        public override DateTime GetStartTimeUtc()
        {
              return this.coreDiagnostics.GetStartTimeUtc();
        }
```